### PR TITLE
[Merged by Bors] - feat: adds `IsSelfAdjoint.algebraMap`

### PR DESCRIPTION
**Pull request by j-loreaux**
`algebraMap R A r` is selfadjiont when `A` is a star `R`-algebra and `r` is selfadjoint.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Star/Module.lean [hunk 1]
```diff
@@ -183,9 +185,21 @@ def StarModule.decomposeProdAdjoint : A ≃ₗ[R] selfAdjoint A × skewAdjoint A
...
 @[simp]
-theorem algebraMap_star_comm {R A : Type*} [CommSemiring R] [StarRing R] [Semiring A]
-    [StarMul A] [Algebra R A] [StarModule R A] (r : R) :
-    algebraMap R A (star r) = star (algebraMap R A r) := by
+theorem algebraMap_star_comm (r : R) : algebraMap R A (star r) = star (algebraMap R A r) := by
   simp only [Algebra.algebraMap_eq_smul_one, star_smul, star_one]
 #align algebra_map_star_comm algebraMap_star_comm
+
+variable (A) in
+protected lemma IsSelfAdjoint.algebraMap {r : R} (hr : IsSelfAdjoint r) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/Star/Module.lean:201]: Is there an `Iff` version of this when `algebraMap` is injective?

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Algebra/Star/Module.lean (hunk 1 above)
Comment (j-loreaux) [reply Mathlib/Algebra/Star/Module.lean:201]: yes, done.

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**